src/main/scala/org/broadinstitute/dsde/firecloud/service/TrialServiceSupport.scala

```diff
@@ -0,0 +1,72 @@
+import org.broadinstitute.dsde.firecloud.model.{UserInfo, WorkbenchUserInfo}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait TrialServiceSupport extends LazyLogging {
+
+  private val dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ")
+
+  def makeSpreadsheetResponse(spreadsheetId: String): SpreadsheetResponse = {
+    SpreadsheetResponse(s"https://docs.google.com/spreadsheets/d/$spreadsheetId")
+  }
+
+  def makeSpreadsheetProperties(title: String): SpreadsheetProperties = {
+    val dateString = dateFormat.format(new Date())
+    new SpreadsheetProperties().setTitle(s"$title: $dateString")
+  }
+
+  // TODO: Look further into the Onix "User Agreement" field
+  def makeSpreadsheetValues(managerInfo: UserInfo, trialDAO: TrialDAO, thurloeDAO: ThurloeDAO, majorDimension: String, range: String)
+    (implicit executionContext: ExecutionContext): Future[ValueRange] = {
+    import scala.collection.JavaConverters._
+    val projects: Seq[TrialProject] = trialDAO.projectReport
+    val trialStatuses = Future.sequence(projects.map { p =>
+      if (p.user.isDefined)
+        thurloeDAO.getTrialStatus(p.user.get.userSubjectId, managerInfo)
+      else
+        Future[Option[UserTrialStatus]](None)
+    })
+    trialStatuses.map { userTrialStatuses =>
+      val headers = List("Project Name", "User Subject Id", "User Email", "Enrollment Date", "Terminated Date", "User Agreement").map(_.asInstanceOf[AnyRef]).asJava
+      val rows: List[util.List[AnyRef]] = projects.map { trialProject =>
+        val (userSubjectId, userEmail, enrollmentDate, terminatedDate) = getTrialUserInformation(trialProject.user, userTrialStatuses)
+        List(trialProject.name.value, userSubjectId, userEmail, enrollmentDate, terminatedDate, "Accepted").map(_.asInstanceOf[AnyRef]).asJava
+      }.toList
+      val values: util.List[util.List[AnyRef]] = (headers :: rows).asJava
+      new ValueRange().setMajorDimension(majorDimension).setRange(range).setValues(values)
+    }
+  }
+
+  // convenience method to pull user information from options
+  private def getTrialUserInformation(user: Option[WorkbenchUserInfo], userTrialStatuses: Seq[Option[UserTrialStatus]]): (String, String, String, String) = {
+    if (user.isDefined) {
+      val userSubjectId = user.get.userSubjectId
+      val userEmail = user.get.userEmail
+      val userTrialStatus = userTrialStatuses.find { trialStatus =>
+        trialStatus.isDefined && trialStatus.get.userId.equals(userSubjectId)
+      }.flatten
+      val enrollmentDate = if (userTrialStatus.isDefined)
+        dateFormat.format(Date.from(userTrialStatus.get.enrolledDate))
+      else
+        ""
+      val terminatedDate = if (userTrialStatus.isDefined)
+        dateFormat.format(Date.from(userTrialStatus.get.terminatedDate))
+      else
+        ""
+      (userSubjectId, userEmail, enrollmentDate, terminatedDate)
+    } else {
+      ("", "", "", "")
+    }
+  }
+
+}
```

Comment:
`TrailServiceSupport` trait seems to exist for Onix spreadsheet reporting purposes only. Does it make sense to rename it to reflect that?

Reply:
The intention is that this supports all of `TrialService`. See `LibraryServiceSupport` and `PermissionsSupport` for how we approach this idea.

Reply:
We also have `Trial.scala` that I thought served the same purpose albeit it's an `object`. Not a big deal, though.

Reply:
The `Trial` class looks like a holder for all of the trial-related case classes/objects - the spreadsheet utilities didn't seem to fit there. If this does become a something more broadly used, it should probably go into a google support class instead of a trial support class.